src/formatDate.js

```diff
@@ -50,7 +50,47 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const data = date
+    .split(fromFormat.slice(-1))
+    .reduce((dateInfo, datePart, i) => {
+      dateInfo[fromFormat[i][0]] = datePart;
+
+      return dateInfo;
+    }, {});
+
+  const newDate = [];
+  const separator = toFormat.slice(-1);
+
+  for (let i = 0; i < toFormat.length - 1; i++) {
+    const dateFormat = toFormat[i];
+    const datePart = data[dateFormat[0]];
+
+    newDate.push(dateFormat[0] === 'Y'
+      ? handleYears(dateFormat, datePart, fromFormat)
+      : datePart);
```

Comment:
You can move this ternary to a datePart variable

Reply:
Oh, yeah, that's right